### PR TITLE
chore: add MsgSendPacket

### PR DESCRIPTION
**Pull request by bznein**
## Description

Huge diff but:

Almost 900 added lines are auto-generated proto stuff
a couple of files have been brought in by https://github.com/cosmos/ibc-go/pull/7238

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

### CODE REVIEW

**Comment by bznein** [comment]
Opening this up as "ready" just to verify that E2E pass. Not ready for review yet

File: modules/core/05-port/keeper/keeper.go [hunk 1]
```diff
@@ -86,5 +87,17 @@ func (k *Keeper) LookupModuleByPort(ctx sdk.Context, portID string) (string, *ca
 // Route returns a IBCModule for a given module, and a boolean indicating
 // whether or not the route is present.
 func (k *Keeper) Route(module string) (types.IBCModule, bool) {
-	return k.Router.GetRoute(module)
+	routes, ok := k.Router.GetRoute(module)
```
Comment (damiannolan) [inline modules/core/05-port/keeper/keeper.go:90]: nit: `routes` -> `route`?

File: modules/core/05-port/keeper/keeper.go [hunk 2]
```diff
@@ -86,5 +87,17 @@ func (k *Keeper) LookupModuleByPort(ctx sdk.Context, portID string) (string, *ca
...
+
+	if ok {
+		return routes, true
+	}
+
+	for _, prefix := range k.Router.Keys() {
+		if strings.Contains(module, prefix) {
```
Comment (damiannolan) [inline modules/core/05-port/keeper/keeper.go:97]: I think we should use `HasPrefix` here rather than doing a contains

**Review by damiannolan — APPROVED** [review]

**Comment by chatton** [comment]
Will address the comments when I handle the merge conflicts from #7238

**Pull request merged**